pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -410,6 +410,19 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 	// Without those ips, it's not possible to execute next calls
 	if !kube.ArePodIPsReady(podList) {
 		reqLogger.Info("Pods IPs are not ready yet")
+		err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: infinispan.Namespace, Name: infinispan.Name}, infinispan)
+		if err != nil {
+			reqLogger.Error(err, "failed to reload Infinispan status")
+			return reconcile.Result{}, err
+		}
+		if infinispan.SetCondition(infinispanv1.ConditionWellFormed, metav1.ConditionUnknown, "Pods are not ready") || infinispan.Status.StatefulSetName != statefulSet.Name {
+			infinispan.Status.StatefulSetName = statefulSet.Name
+			err := r.client.Status().Update(context.TODO(), infinispan)
+			if err != nil {
+				reqLogger.Error(err, "failed to update Infinispan status")
```

Comment:
Should we create a function the handle the updating of the status?

A quick search in `infinispan_controller` shows that before this change there's already 9 places where we log "failed to update Infinispan status"

Reply:
Yeah, I was discussed this case with @rigazilla before and he was not agree with creating common function for update.

Reply:
Yep, my impression is that we are just wrapping the error log in a function without saving a lot. 

When we discussed this the code was more unstable than today, I was also worried to maintain the flexibility to change the error handling for a single situation.

Now I just have a slight preference for inline code because I consider it more readable than a function.
No problem in adding a function anyway, if you think it worth

Reply:
I think we should reconsider. Adding something simple like the following to `infinispan_controller` centralises the logic:

```go
func (r *ReconcileInfinispan) update(ispn *infinispanv1.Infinispan, logger logr.Logger) error {
	if err := r.client.Status().Update(context.TODO(), ispn); err != nil {
		logger.Error(err, "Failed to update Infinispan Status")
		return err
	}
	return nil
}
```

We can then do something like the following when returning from the various reconcile functions:

```
return &reconcile.Result{}, update(ispn, logger)
```

Reply:
does this works in a general way? Sometimes code returns only in case of error.

Reply:
Thanks, @rigazilla and @ryanemerson sure.
As Status update flow is always static
- get latest Infinispan CR
- verify that update is needed
- set values
- update
will try to implement common function for all possible cases